awpy/analytics/map_control.py

```diff
@@ -1,330 +1,368 @@
+
     return tile_to_neighbors
 
 
-def _calc_map_control_helper(
+def calc_parsed_frame_map_control_values(
     map_name: str,
-    current_player_data: dict,
-) -> FrameMapControl:
-    """Helper function to calculate tile map control values for each team.
+    current_player_data: FrameTeamMetadataObject,
+) -> FrameMapControlValues:
+    """Calculate tile map control values for each team given parsed frame.
 
     Args:
-        map_name (string)            : Map used for find_closest_area and
-                                       and relevant tile neighbor dictionary
-        current_player_data (dict)   : Dictionary containing alive player
-                                       locations. Expects extract_player_positions
-                                       output format
+        map_name (str): Map used for find_closest_area and
+                        relevant tile neighbor dictionary
+        current_player_data (FrameTeamMetadataObject): Object containing team metadata
+                                                       (player positions, etc.)
+                                                       Expects extract_player_positions
+                                                       output format
 
-    Returns: Dictionary for each team's map control values
+    Returns: FrameMapControlValues object containing each team's map control values
 
     Raises:
         ValueError: If map_name is not in awpy.data.NAV
-                    If current_player_data not of expected
-                    awpy frame format
+
+    Values are allocated to tiles depending on how many tiles are between it
+    and the source tile (aka tile distance). The smaller the tile distance,
+    the close the tile's value is to 1. The tile distance can get as high as
+    the max_depth argument, which is 10 as default. If the max_depth is kept
+    as default 10, this means the BFS search will stop once a tile
+    distance of 10 is reached.
     """
     if map_name not in NAV:
-        raise ValueError("Map not found.")
-    if "t" not in current_player_data or "ct" not in current_player_data:
-        raise ValueError("Player data dictionary not of expected awpy frame formatting")
+        msg = "Map not found."
+        raise ValueError(msg)
 
-    t_locations = current_player_data["t"]["player-locations"]
-    ct_locations = current_player_data["ct"]["player-locations"]
+    t_locations = current_player_data["t"]["alive_player_locations"]
+    ct_locations = current_player_data["ct"]["alive_player_locations"]
 
-    neighbors_dict = graph_to_tile_neighbors(list(NAV_GRAPHS[map_name].edges))
+    neighboring_tiles = list(NAV_GRAPHS[map_name].edges)
 
-    t_tiles = [find_closest_area(map_name, i)["areaId"] for i in t_locations]
-    ct_tiles = [find_closest_area(map_name, i)["areaId"] for i in ct_locations]
+    neighboring_tile_id_objs = [
+        (TileIdObject(i), TileIdObject(j)) for i, j in neighboring_tiles
+    ]
 
```

Comment:
Probably do not need to assign this to a variable first.

Reply:
Addressed this in next commit to branch.